File: solvers.matlab/gen6p/test_num_gen6p.m
rng(23);
N = 1;
d = 64;

stats = ini_stats('gen6p');

for i = 1:N

    data = inidata_num_gen6p();
    [C,U,dU] = coefs_gen6p(data);

    tic;
    S = red_78x155_gen6p(C);
    stats.tm = [stats.tm toc];
    if isempty(S); continue; end
    
    for k = 1:size(S,2)
        %Rt = getRt6p_gen(data,S(:,k));
        %if norm(Rt-Rtgt,'fro')<1e-5
        %    disp(S(:,k));
        %end
        %disp(S(1,k));
    end

    [stats,~] = update_stats(stats,S,C,U,dU,d,'iterRef',0);

end

disp_stats(stats,'plotHist',1,'saveHist',1);

File: solvers.julia/gen6p/test_num_gen6p.jl
include("../_common/header.jl")
include("inidata_num_gen6p.jl")
include("coefs_gen6p.jl")
include("red_78x155_gen6p.jl")

function test_num_gen6p()

    Random.seed!(23)
    N = 10000
    d = 64

    stats = ini_stats("gen6p")

    for i in 1:N
        data = inidata_num_gen6p()
        C,U = coefs_gen6p(data)

        toc = @elapsed S = red_78x155_gen6p(C)
        stats.tm = [stats.tm; toc]
        if isempty(S); continue; end

        stats = update_stats(stats,S,C,U,d)
    end

    disp_stats(stats)

end

test_num_gen6p()